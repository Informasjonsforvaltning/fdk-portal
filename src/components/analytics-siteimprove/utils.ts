export enum EventCategory {
  DETAILS_PAGE = 'Details page'
}

export enum EventAction {
  REQUEST_ACCESS = 'Request access'
}

type SiteImproveEventProps = {
  category: EventCategory;
  action: EventAction;
  label?: string | undefined;
};

export const trackSiteImproveEvent = ({
  category,
  action,
  label
}: SiteImproveEventProps) => {
  if (window._sz === undefined) {
    // eslint-disable-next-line no-console
    console.error('Unable to find Site Improve event library.');
  } else if (label) {
    window._sz.push(['event', category, action, label]);
  } else {
    window._sz.push(['event', category, action]);
  }
};
