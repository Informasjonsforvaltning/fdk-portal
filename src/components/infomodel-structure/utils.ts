import { InformationModelProperty, TextLanguage } from '../../types';

const applyRelationalTitle = (
  hasObjectType?: string | null,
  title?: Partial<TextLanguage> | null
) => {
  const parentObject = hasObjectType?.split('#').pop();
  const objectString = parentObject
    ? `${parentObject.charAt(0).toUpperCase() + parentObject.slice(1)}: `
    : '';
  return title
    ? Object.entries(title).reduce(
        (previous, [key, value]) => ({
          ...previous,
          [key]: objectString + value
        }),
        {} as Partial<TextLanguage>
      )
    : title;
};

export const resolvePropertySymmetries = (
  modelProperties: Record<string, Partial<InformationModelProperty>>
) =>
  Object.entries(modelProperties).reduce((previous, [key, value]) => {
    if (value.formsSymmetryWith) {
      return {
        ...previous,
        [key]: {
          ...value,
          title: applyRelationalTitle(
            modelProperties[value.formsSymmetryWith].hasObjectType,
            value.title
          )
        },
        [value.formsSymmetryWith]: {
          ...modelProperties[value.formsSymmetryWith],
          formsSymmetryWith: key,
          title: applyRelationalTitle(
            value.hasObjectType,
            modelProperties[value.formsSymmetryWith].title
          )
        }
      };
    }

    return { ...previous, [key]: value };
  }, {} as Record<string, Partial<InformationModelProperty>>);
