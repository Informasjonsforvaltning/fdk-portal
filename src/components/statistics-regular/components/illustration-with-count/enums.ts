export enum Variant {
  COLUMN = 'column'
}
