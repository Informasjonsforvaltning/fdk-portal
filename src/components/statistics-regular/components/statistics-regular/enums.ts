export enum Variant {
  LARGE = 'large',
  XLARGESTRONG = 'xlargeStrong'
}
