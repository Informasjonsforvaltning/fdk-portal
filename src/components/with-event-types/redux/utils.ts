import { Namespace, Fetcher, graph, isLiteral, Literal } from 'rdflib';

import type { EventType } from '../../../types';
import { getConfig } from '../../../config';

const rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#');
const skos = Namespace('http://www.w3.org/2004/02/skos/core#');

export const getEventTypes = async (): Promise<EventType[]> => {
  const fetcher = new Fetcher(graph(), {});

  await fetcher.load(`${getConfig().searchHost.host}/api/events`, {
    withCredentials: false
  });

  const eventTypes = fetcher.store.statementsMatching(
    null,
    rdf('type'),
    skos('Concept')
  );

  return eventTypes.map((item: any) => ({
    uri: item.subject.value ?? '',
    prefLabel: fetcher.store
      .statementsMatching(item.subject, skos('prefLabel'))
      .map(({ object }) => object as Literal)
      .filter(isLiteral)
      .reduce(
        (previous, { value, language }) => ({
          ...previous,
          [language]: value
        }),
        {}
      ),
    broader: fetcher.store
      .statementsMatching(item.subject, skos('broader'))
      .map(({ object }) => object.value)
  }));
};
