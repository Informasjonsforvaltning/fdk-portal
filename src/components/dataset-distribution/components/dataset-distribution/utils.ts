import { DataFormat } from '../../../../types/enums';

const priorities: { [key: string]: number } = {
  [DataFormat.JSON]: 1,
  [DataFormat.XML]: 2,
  [DataFormat.CSV]: 3
};

export const formatSorter = (a: DataFormat, b: DataFormat): number => {
  if (!(a in priorities)) {
    return 1;
  }
  if (!(b in priorities)) {
    return -1;
  }
  return priorities[a] > priorities[b] ? 1 : -1;
};
