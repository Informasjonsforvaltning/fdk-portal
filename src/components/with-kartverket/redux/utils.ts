import {
  Namespace,
  Fetcher,
  graph,
  isLiteral,
  isNamedNode,
  Literal
} from 'rdflib';

import type { AdministrativeUnit } from '../../../types';
import { AdministrativeUnitType } from '../../../types/enums';

const owl = Namespace('http://www.w3.org/2002/07/owl#');
const rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#');
const rdfs = Namespace('http://www.w3.org/2000/01/rdf-schema#');

export const getAdministrativeUnit = async (
  uri: string
): Promise<AdministrativeUnit> => {
  const fetcher = new Fetcher(graph(), { withCredentials: false });

  await fetcher.load(
    `https://rdf.kartverket.no/api/1.0/adminstrative_unit/describe?uri=${uri}&mode=CBD&format=text/turtle`,
    {}
  );

  const subject = fetcher.store.anyStatementMatching(
    null,
    rdf('type'),
    owl('NamedIndividual')
  )?.subject;

  return {
    uri: subject?.value ?? '',
    name: fetcher.store
      .statementsMatching(subject, rdfs('label'))
      .map(({ object }) => object as Literal)
      .filter(isLiteral)
      .reduce(
        (previous, { value, language }) => ({
          ...previous,
          [language]: value
        }),
        {}
      ),
    type: fetcher.store
      .statementsMatching(subject, rdf('type'))
      .map(({ object }) => object)
      .filter(isNamedNode)
      .filter(({ value }) =>
        Object.values(AdministrativeUnitType).includes(value as any)
      )
      .pop()?.value as AdministrativeUnitType
  };
};
