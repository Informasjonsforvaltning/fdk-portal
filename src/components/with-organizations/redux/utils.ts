import { List, Map } from 'immutable';

import { getTranslateText as translate } from '../../../lib/translateText';

import { SortOrder } from '../../../types/enums';

export const sortOrganizations =
  (selector: string[], order: SortOrder = SortOrder.ASC) =>
  (organizations: List<any>) =>
    organizations.sortBy(
      (organization: Map<string, any>) =>
        organization.getIn(selector) ?? organization.get('name'),
      (a?: any, b?: any) =>
        (isNaN(a) && isNaN(b)
          ? translate(Map.isMap(a) ? a.toJS() : a)?.localeCompare(
              translate(Map.isMap(b) ? b.toJS() : b),
              'nb'
            ) ?? 0
          : a - b) * (order === SortOrder.ASC ? 1 : -1)
    );
