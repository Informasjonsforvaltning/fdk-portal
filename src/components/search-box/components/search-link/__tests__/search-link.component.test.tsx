/**
 * @jest-environment jsdom
 */
import React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';

import SearchLink from '../search-link.component';
import { Expectation } from '../../../../../../test/utils';
import { Entity } from '../../../../../types/enums';

afterEach(cleanup);

describe('SearchLink component', () => {
  it(Expectation.STRUCTURE, () => {
    const container = shallow(<SearchLink entity={Entity.DATASET} />);
    expect(container).toMatchSnapshot();
  });
});
