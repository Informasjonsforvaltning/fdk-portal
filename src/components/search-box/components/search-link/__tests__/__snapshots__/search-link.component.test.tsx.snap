// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLink component must render correct structure 1`] = `
<Styled(Link)
  to="/datasets"
  type="dataset"
>
  <Styled(SvgIconCatalogDatasetLg) />
  <span>
    Show only
  </span>
  <span>
    dataset
  </span>
</Styled(Link)>
`;
